Frontend/src/app/components/banner/CookieConsent.tsx

```diff
@@ -0,0 +1,153 @@
+"use client";
+
+import { CookieIcon } from "lucide-react";
+import { Button } from "@/components/ui/button";
+import { useEffect, useState } from "react";
+import { cn } from "@/lib/utils";
+
+import Link from "next/link";
+import { setCookiesAccepted } from "@/lib/cookies";
+import {
+  deleteSessionFromCookies,
+  commitSessionToCookies,
+} from "@/lib/session";
+
+export default function CookieConsent({
+  variant = "default",
+  onAcceptCallback = async () => {
+    await setCookiesAccepted();
+    await commitSessionToCookies();
+  },
+  onDeclineCallback = async () => {
+    try {
+      await deleteSessionFromCookies();
+      console.log("User denied cookie consent");
+    } catch (error) {
+      console.error("Error during handling deny:", error);
+    }
+  },
+}) {
+  const [isOpen, setIsOpen] = useState(false);
+  const [hide, setHide] = useState(false);
+
+  const accept = async () => {
+    setIsOpen(false);
+    document.cookie =
+      "cookieConsent=true; expires=Fri, 31 Dec 9999 23:59:59 GMT";
+    setTimeout(() => {
+      setHide(true);
+    }, 700);
+    await onAcceptCallback();
+  };
+
+  const decline = async () => {
+    setIsOpen(false);
+    setTimeout(() => {
+      setHide(true);
+    }, 700);
+    await onDeclineCallback();
+  };
+
+  useEffect(() => {
+    try {
+      setIsOpen(true);
+      if (document.cookie.includes("cookieConsent=true")) {
```

Comment:
This should likely be `getCookiesAccepted`, we have predefined behaviour to check if the cookie has already been done. Wherever possible, we should aim not to duplicate code.

Reply:
replaced to getCookiesAccepted